### PR TITLE
Fix dynamic symbol computation with gnu hash

### PR DESCRIPTION
**Pull request by ghost**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

When the highest bucket had the index 0, the proper symbol number wasn't
correctly computed. (Ty @ret2libc)

**Test plan**
...

**Closing issues**

...

### CODE REVIEW

File: librz/bin/format/elf/elf_symbols.c [hunk 1]
```diff
@@ -112,6 +110,8 @@ static Elf_(Word) get_index_from_chain(ELFOBJ *bin, ut64 bucket_offset, Elf_(Wor
 
 	while (1) {
 		index++;
+
+		Elf_(Word) tmp;
```
Comment (wargio) [inline librz/bin/format/elf/elf_symbols.c:114]: tmp = 0?

File: librz/bin/format/elf/elf_symbols.c [hunk 2]
```diff
@@ -141,14 +137,17 @@ static Elf_(Word) get_number_of_symbols_from_gnu_hash(ELFOBJ *bin) {
 
 	ut64 pos = hash_offset;
 
+	Elf_(Word) number_of_bucket;
```
Comment (wargio) [inline librz/bin/format/elf/elf_symbols.c:140]: number_of_bucket = 0

File: librz/bin/format/elf/elf_symbols.c [hunk 3]
```diff
@@ -141,14 +137,17 @@ static Elf_(Word) get_number_of_symbols_from_gnu_hash(ELFOBJ *bin) {
...
 	if (!Elf_(rz_bin_elf_read_word)(bin, &pos, &number_of_bucket)) {
 		return 0;
 	}
 
+	Elf_(Word) symbol_base;
```
Comment (wargio) [inline librz/bin/format/elf/elf_symbols.c:145]: symbol_base = 0

File: librz/bin/format/elf/elf_symbols.c [hunk 4]
```diff
@@ -141,14 +137,17 @@ static Elf_(Word) get_number_of_symbols_from_gnu_hash(ELFOBJ *bin) {
...
 	if (!Elf_(rz_bin_elf_read_word)(bin, &pos, &symbol_base)) {
 		return 0;
 	}
 
+	Elf_(Word) bitmask_nwords;
```
Comment (wargio) [inline librz/bin/format/elf/elf_symbols.c:150]: bitmask_nwords = 0

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]
For me is ok, but i really do not like uninitialized variables.

File: librz/bin/format/elf/elf_symbols.c (hunk 4 above)
Comment (ghost) [reply librz/bin/format/elf/elf_symbols.c:150]: It's a RZ_OUT variable.

File: librz/bin/format/elf/elf_symbols.c (hunk 3 above)
Comment (ghost) [reply librz/bin/format/elf/elf_symbols.c:145]: It's a RZ_OUT variable.

File: librz/bin/format/elf/elf_symbols.c (hunk 2 above)
Comment (ghost) [reply librz/bin/format/elf/elf_symbols.c:140]: It's a RZ_OUT variable.

File: librz/bin/format/elf/elf_symbols.c (hunk 1 above)
Comment (ghost) [reply librz/bin/format/elf/elf_symbols.c:114]: It's a RZ_OUT variable.

**Pull request merged**